src/app/vendors/d2-vendors.ts

```diff
@@ -224,10 +225,18 @@ export function filterVendorGroups(
     .filter((g) => g.vendors.length);
 }
 
-export function filterToUnacquired(ownedItemHashes: Set<number>): VendorFilterFunction {
-  return ({ owned, item, collectibleState }) =>
+export function filterToUnacquired(
+  ownedItemHashes: Set<number>,
+  defs: D2ManifestDefinitions | undefined,
+): VendorFilterFunction {
+  return ({ owned, item, collectibleState, failureStrings }) =>
     item &&
     !owned &&
+    !failureStrings.includes(
+      defs?.Vendor.get(specialVendorStrings.alreadyAcquiredFailureString.vendorHash).failureStrings[
+        specialVendorStrings.alreadyAcquiredFailureString.index
+      ] || 'FallbackToPreventBadFiltering',
```

Comment:
Same comment as before - this can cause a crash. What I'm looking for is:

```suggestion
      defs?.Vendor.get(specialVendorStrings.alreadyAcquiredFailureString.vendorHash)?.failureStrings[
        specialVendorStrings.alreadyAcquiredFailureString.index
      ] || 'FallbackToPreventBadFiltering',
```